Enabling module to run from a subdirectory.

robkorobkin opened the pull request:
Hello from Portland, Maine!

I'm a civic hacker working with a couple friends here to set up a data visualization of our municipality's city budget. We love what you guys have built and are looking forward to running our own instance of it.

One issue that I had was that as the app is currently routed, the URLs are all absolute, which would require the component to run at the root directory of our website. That seemed like kind of an annoying constraint. So I made an edit!

It's got three parts to it:

1. CSS image paths now load relative to the CSS file itself.

2. There's now a global variable $app_dir created at the bottom of the config file that strips the path off the page address of the file that we get routed to from the .htaccess, and then this variable, which will remain constant across the deep linking, is used throughout the app for linking between pages and library loading.

3. Lastly, the JS has been rewritten so that instead of doing a string parse of the URI, the PHP just json_encode's the $_GET params and the js runs initialize() off the resulting JS object.

I'd be honored if you'd consider it for submission!

Thanks for your time,

Rob Korobkin
rob.korobkin@gmail.com
617-733-1780

hrothgar commented:
@robkorobkin Thank you Rob! This looks good, but I won't be able to sit down and test it immediately. I'll take care of it next week.

robkorobkin commented:
Hey Hrothgar. No worries. I actually found a bug that needs to get resolved before my fix will work. I'll submit another pull request once that's done. Thanks for getting back to me! Are you part of the group in Arlington that built the original code base? It'd be cool to connect with folks who have experience deploying the platform in other communities, so we can get a sense for what kind of deployment strategy might work best for us here in Portland once we're live?

Ps: Cool Beowulf handle!

robkorobkin commented:
Just pushed in a bug fix. I'm a bit new to pull requests - but I believe you should have it. Let me know if you have time to pull this code in and take it for a spin?

hrothgar commented:
Hey @robkorobkin, thanks again for your efforts here. This is a good feature to have.

I just tested it on my machine (running a local server) and noticed a problem: If I point the webserver root to some parent directory of `src/httpdocs`, then I can navigate to the `httpdocs` folder and everything runs fine. However, if I point the webserver root to `src/httpdocs` itself (as suggested by the current README of this repo), then it does not work! Looking at the page source, it seems that the JavaScript file URLs ended up as, e.g., `//js/home.js` — might this be a special case for the `$app_dir` variable to handle?

I was not part of the team that produced this original code, but I have been working on a V2 of the visual budget project. For information about deployment you should chat with the folks at VisGov (visgov.com), who can certainly answer questions about strategy for that. They'll be excited to hear from you!

robkorobkin commented:
Just checked in a fix. That first slash was getting added twice, once
server side and then again in the client side javascript, so I removed the
place in the JS where the second slash was being added. I think that fixes
it.

Thinking about building an Angular app to offer a web UI for managing the
JSON data files. I figure the CSV import is a great way to import the data
originally, but it could be cool to have a web GUI admin for maintaining it
once it's in. I'd be implementing a no sql solution strictly working off of
the JSON flat file, so it should be fairly straight forward to roll out.
Just an idea, but I'm curious what you think about it?

On Fri, Jan 27, 2017 at 3:05 PM, Hrothgar <notifications@github.com> wrote:

> Hey @robkorobkin <https://github.com/robkorobkin>, thanks again for your
> efforts here. This is a good feature to have.
>
> I just tested it on my machine (running a local server) and noticed a
> problem: If I point the webserver root to some parent directory of
> src/httpdocs, then I can navigate to the httpdocs folder and everything
> runs fine. However, if I point the webserver root to src/httpdocs itself
> (as suggested by the current README of this repo), then it does not work!
> Looking at the page source, it seems that the JavaScript file URLs ended up
> as, e.g., //js/home.js — might this be a special case for the $app_dir
> variable to handle?
>
> I was not part of the team that produced this original code, but I have
> been working on a V2 of the visual budget project. For information about
> deployment you should chat with the folks at VisGov (visgov.com), who can
> certainly answer questions about strategy for that. They'll be excited to
> hear from you!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/goinvo/Visual-Town-Budget/pull/31#issuecomment-275761163>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAy7L4FbcfPABlnpFkflraSF7UaHcHaTks5rWk3-gaJpZM4LlRn_>
> .
>



-- 
----------------------------------------
Rob Korobkin
Portland, ME
617.733.1780

robkorobkin commented:
Just re-read your bug report and added another fix. I've used Git for a
while on my own projects, but I'm kind of new to the whole Pull Requests
world of contributing to open source projects. Let me know if you're able
to get the latest from my repo or if there's a step that I'm missing?

On Mon, Jan 30, 2017 at 5:52 PM, rob korobkin <rob.korobkin@gmail.com>
wrote:

> Just checked in a fix. That first slash was getting added twice, once
> server side and then again in the client side javascript, so I removed the
> place in the JS where the second slash was being added. I think that fixes
> it.
>
> Thinking about building an Angular app to offer a web UI for managing the
> JSON data files. I figure the CSV import is a great way to import the data
> originally, but it could be cool to have a web GUI admin for maintaining it
> once it's in. I'd be implementing a no sql solution strictly working off of
> the JSON flat file, so it should be fairly straight forward to roll out.
> Just an idea, but I'm curious what you think about it?
>
> On Fri, Jan 27, 2017 at 3:05 PM, Hrothgar <notifications@github.com>
> wrote:
>
>> Hey @robkorobkin <https://github.com/robkorobkin>, thanks again for your
>> efforts here. This is a good feature to have.
>>
>> I just tested it on my machine (running a local server) and noticed a
>> problem: If I point the webserver root to some parent directory of
>> src/httpdocs, then I can navigate to the httpdocs folder and everything
>> runs fine. However, if I point the webserver root to src/httpdocs itself
>> (as suggested by the current README of this repo), then it does not work!
>> Looking at the page source, it seems that the JavaScript file URLs ended up
>> as, e.g., //js/home.js — might this be a special case for the $app_dir
>> variable to handle?
>>
>> I was not part of the team that produced this original code, but I have
>> been working on a V2 of the visual budget project. For information about
>> deployment you should chat with the folks at VisGov (visgov.com), who
>> can certainly answer questions about strategy for that. They'll be excited
>> to hear from you!
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/goinvo/Visual-Town-Budget/pull/31#issuecomment-275761163>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe-auth/AAy7L4FbcfPABlnpFkflraSF7UaHcHaTks5rWk3-gaJpZM4LlRn_>
>> .
>>
>
>
>
> --
> ----------------------------------------
> Rob Korobkin
> Portland, ME
> 617.733.1780 <(617)%20733-1780>
>



-- 
----------------------------------------
Rob Korobkin
Portland, ME
617.733.1780

hrothgar commented:
@robkorobkin An existing pull request will automatically update with each new commit you make to your fork. I can see your changes. Will take a look today or tomorrow — thank you!

Did you get in touch with Visgov?

craigmcginley commented:
Hi Rob,

It would be helpful to have updated docs/instructions to accompany the changes. From what I can tell, this PR appears to modify the project to be run only as a module and not a standalone package. The for this project would be to keep it as an all-in-one tool to instantiate a visual budget instance, even if it is unfortunately not very flexible.

Have you checked out our more recent effort in developing a [wp plugin](https://github.com/goinvo/visualbudget)?

Since your fork is also open source, we could consider linking to it in the docs for users looking for the use case closer to your own.

Please shoot me an email if you want to discuss the goals further, we could set up a brief phone call. craig@goinvo.com

Thanks,
Craig
